src/containers/Booklist.js

```diff
@@ -1,34 +1,48 @@
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import Book from '../components/Book';
+import REMOVE_BOOK from '../reducers/books';
 
 const BookList = (props) => {
-  const { books } = props;
+  const { books, removebook } = props;
+
+  const handleRemoveBook = (book) => {
+    const bookDelete = books.find((bookDel) => book.id === bookDel.id);
+    removebook(bookDelete);
+  };
 
   return (
     <table>
-      <tr>
-        <th>Id</th>
-        <th>Title</th>
-        <th>Category</th>
-      </tr>
-      <tr>
-        {books.map((book) => (<Book key={book.id} book={book} />)) }
-      </tr>
+      <thead>
+        <tr>
+          <th>Id</th>
+          <th>Title</th>
+          <th>Category</th>
+        </tr>
+      </thead>
```

Comment:
This code already exists in the `Book` component. Please keep your code DRY.
The `BookList` could be as simple as:
```react
const BookList = (props) => {
const handleRemoveBook = (book)
 => {...}

return <Book key{book.id} book={book} handleRemoveBook={this.handleRemoveBook} />
}
```
Now the `Book` component could look like:
```react
const Book = (props) => {
render() {
    const book = this.props.book;

    return(
      <tr>
        <td>{this.props.book.id}</td>
        <td>{this.props.book.title}</td>
        <td>{this.props.book.category}</td>
        <td>
          <button onClick={() => this.props.handleBookRemove(book)}>Remove Book</button>
        </td>
      </tr>
    )
  }

}
```
Please make the necessary changes

Reply:
I don't think is the same. This is no stateless component so I shouldn't use this